2012-05-23  H.J. Lu  <hongjiu.lu@intel.com>

	PR ld/13909
	* elf-eh-frame.c (_bfd_elf_maybe_strip_eh_frame_hdr): Ignore
	.eh_frame section created by linker.

diff --git a/bfd/elf-eh-frame.c b/bfd/elf-eh-frame.c
index 685540c..32714bb 100644
--- a/bfd/elf-eh-frame.c
+++ b/bfd/elf-eh-frame.c
@@ -1280,6 +1280,7 @@ _bfd_elf_maybe_strip_eh_frame_hdr (struct bfd_link_info *info)
 	o = bfd_get_section_by_name (abfd, ".eh_frame");
 	while (o != NULL
 	       && (o->size <= 8
+		   || (o->flags & SEC_LINKER_CREATED) != 0
 		   || bfd_is_abs_section (o->output_section)))
 	  o = bfd_get_next_section_by_name (o);
 	if (o != NULL)
